src/lib/gradientParser.ts

```diff
@@ -0,0 +1,223 @@
+  const ret: Node[][] = []
+  let i = 0
+  for (let j = 0; j < nodes.length; j++) {
+    if (nodes[j].type === "div") {
+      ret.push(splitSpaceArgs(nodes.slice(i, j)))
+      i = j + 1
+    }
+  }
+  ret.push(splitSpaceArgs(nodes.slice(i)))
+  return ret
+}
+
+export function parseGradient(css: string): GradientNode[] {
+  const parsed = parse(css)
+  const gradients = splitCommaArgs(parsed.nodes)
+  return gradients
+    .filter(
+      (nodes) =>
+        nodes.length === 1 && nodes[0].type === "function" && nodes[0].value.includes("-gradient")
+    )
+    .map(([node]) => {
+      const args = splitCommaArgs((node as FunctionNode).nodes)
+      const type = node.value.replace(/^-webkit-/, "")
+      if (type.includes("linear-gradient")) {
+        const ret: Partial<LinearGradient> = {
+          type: type as LinearGradient["type"]
+        }
+        if (args[0][0].value === "to") {
+          ret.orientation = {
+            type: "directional",
+            value: args
+              .shift()!
+              .slice(1)
+              .map((it) => it.value)
+              .join(" ") as DirectionalNode["value"]
+          }
+        } else {
+          const first = unit(args[0][0].value)
+          if (first && ["deg", "turn", "rad", "grad"].includes(first.unit)) {
+            ret.orientation = {
+              type: "angular",
+              value: toDegrees(first)
+            }
+            args.shift()
+          } else {
+            ret.orientation = {
+              type: "angular",
+              value: 180
+            }
+          }
+        }
+        ret.colorStops = args.map(toColorStopOrHint)
+        return ret as LinearGradient
+      } else if (type.includes("radial-gradient")) {
+        const ret: Partial<RadialGradient> = {
+          type: type as RadialGradient["type"],
+          endingShape: "ellipse",
+          size: "farthest-corner",
+          position: "center"
+        }
+        let hasShape = false
+        for (let i = 0; i < args[0].length; i++) {
+          const arg = args[0][i]
+          switch (arg.value) {
+            case "circle":
+            case "ellipse":
+              hasShape = true
+              ret.endingShape = arg.value
+              break
+            case "closest-corner":
+            case "closest-side":
+            case "farthest-corner":
+            case "farthest-side":
+              hasShape = true
+              ret.size = arg.value
+              break
+            case "at":
+              hasShape = true
+              ret.position = args[0]
+                .slice(i + 1)
+                .map((it) => stringify(it))
+                .join(" ")
+              break
+          }
+          // TODO dimension size
+        }
+        if (hasShape) args.shift()
+        ret.colorStops = args.map(toColorStopOrHint)
+        return ret as RadialGradient
+      }
+      // TODO conic-gradient()
+      throw new Error("Unsupported gradient: " + stringify(node))
```

Comment:
it would be nice just to parse the pieces of it

Reply:
done